tests/test_text_cleaning.py

```diff
@@ -0,0 +1,413 @@
+        self.assertIn("italic", result)
+        self.assertIn("List item", result)
+        self.assertIn("Sub-item", result)
+
+
+class TestTextCleaningEdgeCases(unittest.TestCase):
+    """Test edge cases and complex scenarios in text cleaning."""
+    
+    def setUp(self):
+        """Set up test fixtures."""
+        self.parser = AttractionsParser()
+    
+    def test_nested_templates(self):
+        """Test cleaning with nested templates."""
+        text = "Text ({{convert|{{lang|en|100}}|m}}) here"
+        result = self.parser._apply_text_cleaning_regex(text)
+        self.assertNotIn("({{convert|", result)
+        self.assertIn("Text", result)
+        self.assertIn("here", result)
+    
+    def test_multiple_galleries(self):
+        """Test multiple gallery tags."""
+        text = """
+        <gallery>
+        Image1.jpg
+        </gallery>
+        
+        Content.
+        
+        <gallery caption="Second">
+        Image2.jpg
+        </gallery>
+        """
+        result = self.parser._remove_gallery_tags(text)
+        self.assertNotIn("<gallery>", result)
+        self.assertNotIn("<gallery caption=", result)
+        self.assertIn("Content", result)
+    
+    def test_mixed_quotes_and_brackets(self):
+        """Test complex quote and bracket combinations."""
+        text = "[['''Bold link''']] and [[''Italic link'']]"
+        result = self.parser._cleanup_excessive_quotes(text)
+        self.assertIn("['Bold link']", result)
+        self.assertIn("['Italic link']", result)
+    
+    def test_empty_inputs(self):
+        """Test handling of empty inputs."""
+        empty_inputs = ["", None, "   ", "\n\n\n"]
+        
+        for empty_input in empty_inputs:
+            with self.subTest(empty_input=empty_input):
+                if empty_input is not None:
+                    result = self.parser._apply_text_cleaning_regex(empty_input)
+                    self.assertIsInstance(result, str)
+    
+    def test_unicode_edge_cases(self):
+        """Test Unicode edge cases."""
+        text = "Hello привіт 你好 مرحبا こんにちは"
+        result = self.parser._apply_text_cleaning_regex(text)
+        self.assertIn("Hello", result)
+        # Non-Latin characters should be filtered out if Latin filtering is enabled
+        # This depends on the parser's configuration
+    
+    def test_template_edge_cases(self):
+        """Test edge cases with templates."""
+        test_cases = [
+            ("({{}})", ""),  # Empty template
+            ("({{template}})", ""),  # Template without params
+            ("({{template|}})", ""),  # Template with empty param
+            ("({{template|param1|param2}})", ""),  # Multiple params
+        ]
+        
+        for input_text, expected in test_cases:
+            with self.subTest(input_text=input_text):
+                result = self.parser._remove_template_parentheses(input_text)
+                self.assertEqual(result, expected)
+
```

Comment:
- For template edge cases, add more malformed templates and parenthesis combinations.
- Add tests for extremely long strings or very large templates to check performance and memory handling.